### PR TITLE
Update README.md (about PG branches and license)

### PR DESCRIPTION
**Pull request by erikmd**
Hi @cpitclaudel @Matafou 

I'm opening this PR to follow-up our latest e-mail discussion.

If the phrasing is OK for you, I can merge this tomorrow (in master + async modulo the merge conflict).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,29 @@
...
+  [async](https://github.com/ProofGeneral/PG/tree/async) branch, and
+  licensed under GPLv3+.
+
+Contributions to this repository are placed under the BSD-3 license.
+As BSD-3 is compatible with both GPLv2 and GPLv3+, this means that
+we can merge them in both master and async branches if need be,
+using the same license as the rest of the codebase, while you keep
+all the rights on your code.
+For more info, see <https://opensource.org/licenses/BSD-3-Clause>.
+
```
Comment (cpitclaudel) [inline README.md]: I'd move this after the overview, and I'd add `experimental` to the description of the async branch.  I'd also move the license text to the end, maybe under a `# Contributing` header

Comment (Matafou) [reply README.md]: +1

**Comment by erikmd** [comment]
Thanks! I Updated the PR as suggested.
@Matafou there is still a brief mention of the license of the branches in the "About Proof General branches" section; is it OK like this or should I move it to the "Contributing" section?

**Comment by Matafou** [comment]
Looks good as it is now.

**Pull request merged**